compaction/compaction_manager.cc

```diff
@@ -536,7 +539,7 @@ requires std::is_base_of_v<compaction_task_executor, TaskExecutor> &&
 requires (compaction_manager& cm, Args&&... args) {
     {TaskExecutor(cm, std::forward<Args>(args)...)} -> std::same_as<TaskExecutor>;
 }
-future<compaction_manager::compaction_stats_opt> compaction_manager::perform_compaction(std::optional<tasks::task_info> parent_info, Args&&... args) {
+future<compaction_manager::compaction_stats_opt> compaction_manager::perform_compaction(bool throw_on_stop, std::optional<tasks::task_info> parent_info, Args&&... args) {
```

Comment:
please define respective bool_class to make the parameter more expilcit in the call sites.

Reply:
reuse throw_if_stopping?